src/main/scala/strawman/collection/immutable/SortedSet.scala

```diff
@@ -0,0 +1,21 @@
+package strawman.collection.immutable
+
+import strawman.collection.{SetMonoTransforms, Sorted, SortedLike}
+
+/** Base trait for sorted sets */
+trait SortedSet[A]
+  extends Set[A]
+    with Sorted[A]
+    with SortedSetLike[A, SortedSet] // Inherited SortedSet operations return a `SortedSet`
+
+trait SortedSetLike[A, +C[X] <: SortedSet[X]]
+  extends SortedLike[A, C]
+    with SetLike[A, Set] // Inherited Set operations return a `Set`
+    with SetOps[A, C[A]] // Override the return type of Set ops to return C[A]
+    with SetMonoTransforms[A, C[A]] {
+
+  // We have to write again the definition of methods inherited from SortedPolyTransforms
+  // so that they take precedence over the one inherited from IterablePolyTransforms
+  def map[B](f: A => B)(implicit ordering: scala.Ordering[B]): C[B]
+
+}
```

Comment:
Probably, but this is not what we want: we really want `SortedSet[A]` to extend `SortedPolyTransforms[A, SortedSet]` and `IterablePolyTransforms[A, Set]` (note that we have `Set` in the latter case).

We could define `SortedPolyTransforms` as follows, as a workaround:

~~~ scala
trait SortedPolyTransforms[A, +C[X] <: Sorted[X], +D[X] <: Iterable[X]]
  extends IterablePolyTransforms[A, D]
~~~